jobs/sendEmailAlertNoFill.php

```diff
@@ -89,28 +88,32 @@ function sendEmail($email, $lastTaskDate, $from, $cc, $subject, $template) {
             $login = $user->getLogin();
             $email = $login . "@" . $COMPANY_DOMAIN;
             $from = $NO_FILL_EMAIL_FROM;
-            $lastTaskDate = TasksFacade::getLastTaskDate($user, $today);
-            if (is_null($lastTaskDate)) {
-                // User never saved tasks, use their contract start date as reference
-                $lastTaskDate = $period[0]->getInitDate();
+            $lastMonday = new DateTime(date("Y-m-d", strtotime("last week monday")));
+            $lastFriday = new DateTime(date("Y-m-d", strtotime("last week friday")));
+
+            $emptyDaysLastWeek = TasksFacade::getEmptyDaysInPeriod($user, $lastMonday, $lastFriday);
+
+            if (empty($emptyDaysLastWeek)) {
+                // User doesn't have any empty day last week
+                continue;
             }
-            $difference = $lastTaskDate->diff($today);
+            $difference = $lastFriday->diff($today);
             $difference = $difference->format('%a');
             if ($difference == $NO_FILL_DAYS_TRIGGER_LAST) {
```

Comment:
A less drastic solution would be to check the user's last date (`$lastTaskDate = TasksFacade::getLastTaskDate($user, $today);`) to decide which message should be sent. The three messages should be rewritten, though, and they should include the `$lastTaskDate` as extra information - it doesn't make much sense to tell a person who hadn't filled phpreport for two or three weeks that they skipped Monday, Tuesday, etc. last week.

Reply:
The whole logic changed indeed, we don't want to check the last time the user updated anymore.

The degrees of severity still exist but is a bit different now, I questioned myself if I should still used these variables or not, maybe this will need to be hardcoded, but the end goal is to have 3 reminders that will be sent during the week that checks if the user completed the whole previous week.

Reply:
I understand! In that case, I would modify the default values for `$NO_FILL_DAYS_TRIGGER_*` in `config.php` to a value that makes sense now, and add a comment to warn people not to use values bigger than 7.